architecture.md

```diff
@@ -0,0 +1,91 @@
+Plugins let Ignition Gazebo utilize other libraries.
+For example, [Ignition Physics](https://ignitionrobotics.org/libs/physics) and Ignition Gazebo are independent of one another, so Ignition Gazebo has a physics plugin that uses Ignition Physics as a library and incorporates Gazebo specifics, allowing Physics to be a system running in the simulation loop.
+
+The Ignition Physics library is only used in the physics plugin, not in other plugins nor in the core of Ignition Gazebo.
+Some libraries are only used by one plugin, or one in each process (frontend and backend).
+However, some foundational libraries are used by all plugins, like [Ignition Common](https://ignitionrobotics.org/libs/common) which provides common functionality like logging, manipulating strings, file system interaction, etc., to all plugins.
+Other such libraries include [Ignition Plugin](https://ignitionrobotics.org/libs/plugin), [Ignition Math](https://ignitionrobotics.org/libs/math), [SDFormat](https://ignitionrobotics.org/libs/sdformat), and more.
+
+There are certain plugins in both the frontend and backend processes that are loaded by default in every version of Ignition.
+Many other plugins are optional.
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+These demos on [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) and [GUI Configuration](https://ignitionrobotics.org/api/gazebo/4.3/gui_config.html) showcase that functionality.
+
+The simulation process is depicted in the diagram below, and further explained in the Backend and Frontend process sections that follow.
+
+![Ignition Gazebo architecture diagram](images/IgnitonGazeboArchicteture.png)
+
+## Backend server process
+
+Ignition Gazebo is responsible for loading plugins in the backend, referred to as systems.
+The server runs an entity-component system architecture (see [Ignition Gazebo terminology](https://ignitionrobotics.org/api/gazebo/4.2/terminology.html)).
+The backend will usually have multiple systems responsible for everything in the simulation – computing physics, recording logs, receiving user commands, etc.
+
+Systems act on entities and components of those entities.
+An entity is anything in the simulation scene (a model, link, light, actor, etc.), and components are their characteristics (pose, name, geometry, etc.).
+Server plugins have access to all entities and what components they have, and make modifications to those components.
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Other Ignition Libraries, however, don’t know about the entities and components.
```

Comment:
I think the last sentence of this paragraph might be a little confusing with respect to the other sentences/content of this paragraph. We give an example of the physics system (plugin) operating on an entity, which might make readers think that `Ignition Physics` has direct access to the entity component system, but then finish the paragraph by saying that Ignition libraries don't know about entities and components. So, would it be better to instead say something like:
```
The entity component system is self-contained in Ignition Gazebo, and systems act on entities and their components.
Since other Ignition Libraries don't have direct knowledge/access to entities and components, other Ignition Libraries should be used in systems so that the server plugin can share entities (and their components) with these other Libraries.
For example, the physics system shares entities and components with the Ignition Physics library in order to achieve physics effects on entities and components.
```

If the assumptions I made in the re-phrase above are incorrect, please let me know - I am still relatively new to Ignition :upside_down_face: This is just a thought/suggestion I had. Feel free to modify it in whatever way that makes the most sense!